src/DotLiquid/Context.cs

```diff
@@ -336,7 +336,9 @@ private object Resolve(string key, bool notifyNotFound = true)
             // Integer.
             match = IntegerRegex.Match(key);
             if (match.Success)
-                return Convert.ToInt32(match.Groups[1].Value);
+            {
+                return Convert.ToInt64(match.Groups[1].Value);

```

Comment:
Here we should have tried backward compatible Int32 first and only on overflow expand to Int64.